bindings/wasm/src/account/storage/traits.rs

```diff
@@ -0,0 +1,307 @@
+  async fn chain_state(&self, did: &IotaDID) -> AccountResult<Option<ChainState>> {
+    let promise: Promise = Promise::resolve(&self.chain_state(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Set the chain state of the identity specified by `did`.
+  async fn set_chain_state(&self, did: &IotaDID, chain_state: &ChainState) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.set_chain_state(did.clone().into(), chain_state.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Returns the state of the identity specified by `did`.
+  async fn state(&self, did: &IotaDID) -> AccountResult<Option<IdentityState>> {
+    let promise: Promise = Promise::resolve(&self.state(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Sets a new state for the identity specified by `did`.
+  async fn set_state(&self, did: &IotaDID, state: &IdentityState) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.set_state(did.clone().into(), state.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Removes the keys and any state for the identity specified by `did`.
+  async fn purge(&self, did: &IotaDID) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.purge(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+}
+
+#[wasm_bindgen(typescript_custom_section)]
+const STORAGE: &'static str = r#"
+/** All methods an object must implement to be used as an account storage. */
+interface Storage {
+  /** Sets the account password.*/
+  setPassword: (encryptionKey: Uint8Array) => Promise<void>;
+
+  /** Write any unsaved changes to disk.*/
+  flushChanges: () => Promise<void>;
+
+  /** 
+   * Attempt to obtain the exclusive permission to modify the given `did`.
+   * The caller is expected to make no more modifications after the lease has been dropped.
+   * Returns an error if already leased.
+  */
+  leaseDid: (did: DID) => Promise<DIDLease>;
+
+  /** Creates a new keypair at the specified `location`.*/
+  keyNew: (did: DID, keyLocation: KeyLocation) => Promise<string>;
+
+  /** Inserts a private key at the specified `location`.*/
+  keyInsert: (did: DID, keyLocation: KeyLocation, privateKey: string) => Promise<string>;
+
+  /** Returns `true` if a keypair exists at the specified `location`.*/
+  keyExists: (did: DID, keyLocation: KeyLocation) => Promise<boolean>;
+
+  /** Retrieves the public key at the specified `location`.*/
+  keyGet: (did: DID, keyLocation: KeyLocation) => Promise<string>;
+
+  /** Deletes the keypair specified by `location`.*/
+  keyDel: (did: DID, keyLocation: KeyLocation) => Promise<void>;
```

Comment:
Is `keyDel` supposed to be idempotent? This came up when I looked over the memory_storage.ts implementation, which behaved slightly differently from the Rust `MemStore`. Idempotency seems like a good idea to me here, and we should state that in the doc as well, then. That would also require changes to all implementations. Feel free to leave this for a later PR, in which case we should open an issue, so it's not forgotten about. I can open the issue if there is agreement for that.